### ISSUE TITLE
Further refine the scoreboard

### ISSUE THREAD

**Issue by Fisico** [teeworlds/teeworlds#1826]
The scoreboard needs some more love.

A quick mockup to highlight, every other line and also highlight the most important Column "Score"
![image](https://user-images.githubusercontent.com/329471/49699075-d463a380-fbcc-11e8-8b26-eacdc1498bcb.png)

Yea the font is not 1:1 as in Teeworlds, I know, but I can’t replicate it. The font rendering is good as it is now.

Scoreboard background container: Black - 30% alpha
Column header: Black - 30% alpha
First line: Black - 10% alpha
Second line: Black - 25% alpha
Score highlight: White – 30% alpha
Player highlight: White – 40% alpha (Fontcolor inverted)

Also there should be a minimum height for the scoreboard, which corresponds to the maximum allowed player slots (spectator slots excluded).
Max players 16 | Spec slots 8 | Gametype TEAM = 4 scoreboard rows
Max players 16 | Spec slots 8 | Gametype Solo = 8 scoreboard rows
Max players 16 | Spec slots 4 | Gametype Team = 6 scoreboard rows

The scoreboard container should expand dynamically.

**Comment by 1stDecadeTW** [comment]
I totally agree, but keep in mind that the scoreboard must be "eye friendly" because is a frecuently access feature.

If you want to keep the 0.7 neat, beauty, full of details and revamped, give the users the option to use the classic and eye friendly 0.6 scoreboard, instead.

With "eye friendly" I mean that there shouldn't be much color or transparent contrast between the game and the scoreboard, maybe I am wrong?.

**Comment by Fisico** [comment]
I don't want a whole new scoreboard.... 
I think my mockup is more eye friendly than the current one. I tried to polish it and make it a bit better, changes some things, which are currently missing or feeling wrong.

I don't think the 0.6 scoreboard is any better.

**Comment by sepulzera** [comment]
In case it looks ingame like in your mockup, here my two cents:
1. The backgrounds are too transparent, making the text very hard to read. Especially the Ping, K and D columns. We should consider to comply with the *web accessibility contrast rules*. That can be verified using tools like http://accessible-colors.com/.
2. The header column should be more emphasized. E. g. with a different background, different font color or different font style (bold).

Furthermore, imo the team player count should be reworked to make it more evident.
Here the screenshot from the above referenced issue:

![49052311-e860db80-f1b0-11e8-903b-25eddc162255](https://user-images.githubusercontent.com/43857716/49723734-da5a9280-fc67-11e8-80c4-d37d86e545a0.png)

Finally, I would like to swap *Clan* with *PlayerName* for the **spectators**. Then it matches the order shown in the scoreboard, PlayerName and then Clan.

**Comment by 1stDecadeTW** [comment]
Get rid of the player count on the header, the "(0)" isn't informative and the team score must be prioritized. 

My suggestion is to move the player count at the bottom and make It look like this, with less font height:

1/16 players.
1/32 players.
1/64 players.
(Depending of the server configuration).

**Comment by Fisico** [comment]
> In case it looks ingame like in your mockup, here my two cents:
> 
> 1. The backgrounds are too transparent, making the text very hard to read. Especially the Ping, K and D columns. We should consider to comply with the _web accessibility contrast rules_. That can be verified using tools like http://accessible-colors.com/.
> 2. The header column should be more emphasized. E. g. with a different background, different font color or different font style (bold).
> 
> Furthermore, imo the team player count should be reworked to make it more evident.
> Here the screenshot from the above referenced issue:
> 
> ![49052311-e860db80-f1b0-11e8-903b-25eddc162255](https://user-images.githubusercontent.com/43857716/49723734-da5a9280-fc67-11e8-80c4-d37d86e545a0.png)
> 
> Finally, I would like to swap _Clan_ with _PlayerName_ for the **spectators**. Then it matches the order shown in the scoreboard, PlayerName and then Clan.

@sepulzera thank you for your feedback!
As I said the text rendering is wrong… you can see the text in the game way better.

I made the scoreboard darker, to match the current one…. New values:
Scoreboard background container: Black - 30% alpha
Column header: Black - 50% alpha
First line: Black - 30% alpha
Second line: Black - 40% alpha
Score highlight: White – 40% alpha
Player highlight: White – 30% alpha (Fontcolor inverted)

I think the team player count is ok as is.

For the spectators, yea I also would change that.

**Comment by sepulzera** [comment]
@1stDecadeTW  @Fisico I moved the discussion about the player count to a separate issue to don't spam this one, as it is imo a separate step.

Back to this one:

> As I said the text rendering is wrong… you can see the text in the game way better.

That makes it very difficult to judge the readibility, imo.

For better comparability here is a screenshot of the current scoreboard.
![screenshot_old_scoreboard](https://user-images.githubusercontent.com/43857716/49753109-a789bc00-fcb2-11e8-9edb-35bcd5ab00d8.jpg)

Compared to your version, I think without your outer rounded corners of the score lines it looks much smoother.

Furthermore, I would remove following corners aswell, though this is highly subjective I guess:
![screenshot_old_scoreboard_marked](https://user-images.githubusercontent.com/43857716/49753360-6d6cea00-fcb3-11e8-8efc-f6a0f35127ba.jpg)

**Comment by Zatline** [comment]
@Fisico I personally don't like any of these mockups not a fan of these zebra stripes u added in, not rly helpful in that narrow designed form it's not like people lose track of rows all day. It looks better and more clean without KISS principle .. 
I'm also not sure about the capital lettering of blue and red it feels out of balance.
and team score in the header should be kept aligned to the center.

I really think moving score at the last column and ping first like cs:go should do the magic.
and perhaps upper overall transparency a bit.

**Comment by Fisico** [comment]
A new mockup without "zebra stripes" and the feedback from @sepulzera  
i think it is quite nice to have all score points in one column:
![image](https://user-images.githubusercontent.com/329471/49759257-71a00400-fcc1-11e8-9ec4-55eff92b6022.png)

![image](https://user-images.githubusercontent.com/329471/49759348-b3c94580-fcc1-11e8-9f81-8a41715eb413.png)

**Comment by Fisico** [comment]
last one for today.... This keeps the feeling from the original scoreboard but also adds some improvements.
![image](https://user-images.githubusercontent.com/329471/49759786-ef184400-fcc2-11e8-9101-d9206f69bf36.png)

**Comment by sepulzera** [comment]
I like the last one most. Btw. CS:GO blurs the background, too. Never noticed, lol.
But it really increases the readability ALOT. 
Maybe we should try to reduce the transparency as alternative to blur.

For the different row colors, I have to agree with @Zatline. Without looks much simpler and cleaner.

Some more things I would change:
- Remove the round corners from the score column inside of the boxes.
- Do not highlight the Score title
- Maybe try even a version without round corners in the score header row + spectators header

**Comment by Zatline** [comment]
@sepulzera 

- round conors are an initial part of the design and teestyle
- would agree with that one
- don't increase the gaps
- as said before

**Comment by Dune-jr** [comment]
The blur is something used a lot in video games to change the focus when you open menus / scoreboard. That would be ideal imo, but not feasible yet.

I don't think stripes or highlights work at all, in the scoreboard.

**Comment by 1stDecadeTW** [comment]
Lets bring the Blur to teeworlds, It look de awesome in the scorebord and might have other uses 😁😍

**Comment by Fisico** [comment]
So some points we can agree on:
- The darker scoreboard background is good.
- The dark column header is better as the light one (brings more consistency to the scoreboard).
- No score column highlight.
- Remove team player indicator (1)
- If possible, blur the background (i think this is not possible with opengl2, no shaders)
- Keep the round corners. (I really like my idea to round of the bottom corner from the header, separates it way better).
![image](https://user-images.githubusercontent.com/329471/49821665-4bd73580-fd7b-11e8-8ad7-2e61dd6dff62.png)

**Comment by Fisico** [comment]
Less transparency and a bit more spacing.

![image](https://user-images.githubusercontent.com/329471/49827985-fd319780-fd8a-11e8-8ae6-3efee8abd4d8.png)

**Comment by sepulzera** [comment]
Love the larger gaps. Please make this happen!
